zebra-consensus/src/checkpoint.rs

```diff
@@ -3,315 +3,1217 @@
-        // TODO(teor || jlusby): check error kind
-        ready_state_service
-            .call(zebra_state::Request::GetBlock {
-                hash: block415000.as_ref().into(),
-            })
-            .await
-            .unwrap_err();
-
-        /// Make sure the state service is ready (2/2)
-        let ready_state_service = state_service.ready_and().await.map_err(|e| eyre!(e))?;
-        /// Make sure neither block is in the state: expect GetBlock 0 to fail.
-        // TODO(teor || jlusby): check error kind
-        ready_state_service
-            .call(zebra_state::Request::GetBlock {
-                hash: block0.as_ref().into(),
-            })
-            .await
-            .unwrap_err();
+        assert_eq!(
+            checkpoint_verifier.previous_checkpoint_height(),
+            BeforeGenesis
+        );
+        assert_eq!(
+            checkpoint_verifier.target_checkpoint_height(),
+            WaitingForBlocks
+        );
+        assert_eq!(checkpoint_verifier.max_checkpoint_height(), BlockHeight(0));
 
         Ok(())
     }
 
     #[tokio::test]
     #[spandoc::spandoc]
-    async fn checkpoint_wrong_hash_fail() -> Result<(), Report> {
-        install_tracing();
+    async fn wrong_checkpoint_hash_fail() -> Result<(), Report> {
+        zebra_test::init();
 
-        let block0 =
+        let good_block0 =
             Arc::<Block>::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])?;
+        let good_block0_hash: BlockHeaderHash = good_block0.as_ref().into();
+        let mut bad_block0 =
+            Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])?;
+        // Change the header hash
+        bad_block0.header.version = 0;
+        let bad_block0: Arc<Block> = bad_block0.into();
 
-        // Make a checkpoint list containing the genesis block height,
-        // but use the wrong hash
-        let genesis_checkpoint_list: HashMap<BlockHeight, BlockHeaderHash> =
-            [(block0.coinbase_height().unwrap(), BlockHeaderHash([0; 32]))]
+        // Make a checkpoint list containing the genesis block checkpoint
+        let genesis_checkpoint_list: BTreeMap<BlockHeight, BlockHeaderHash> =
+            [(good_block0.coinbase_height().unwrap(), good_block0_hash)]
                 .iter()
                 .cloned()
                 .collect();
 
-        let mut state_service = Box::new(zebra_state::in_memory::init());
-        let mut checkpoint_verifier = super::init(state_service.clone(), genesis_checkpoint_list);
+        let mut checkpoint_verifier =
+            CheckpointVerifier::new(genesis_checkpoint_list).map_err(|e| eyre!(e))?;
 
-        /// Make sure the verifier service is ready
+        assert_eq!(
+            checkpoint_verifier.previous_checkpoint_height(),
+            BeforeGenesis
+        );
+        assert_eq!(
+            checkpoint_verifier.target_checkpoint_height(),
+            WaitingForBlocks
+        );
```

Comment:
TIOLI: This setup code seems to be repeated in all of our tests, could we abstract this into a helper function to cut down on the repetition?

Reply:
I've added this refactor to the list in #610.